pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -107,7 +99,9 @@ def __init__(
         if sampled_seq.max_duration == 0:
             raise ValueError("SequenceSamples is empty.")
         # Check compatibility of register and device
+        self._sampling_rate = sampling_rate
         device.validate_register(register)
+        self._device = device
         self._register = register
```

Comment:
You store these here and in `HamiltonianData`, which opens the door to divergence between the two. How about turning these into properties that access the value from `HamiltonianData` instead?